whylog/tests/tests_teacher/__init__.py

```diff
@@ -95,6 +99,57 @@ def test_default_user_parser(self):
         )  # yapf: disable
         assert wanted_effect_parser == effect_parser
 
+    def test_setting_parser_name(self):
+        effect_parser_name = self.teacher.get_rule().parsers[self.effect_id].pattern_name
+        new_name = effect_parser_name + '_hello'
+        self.teacher.set_pattern_name(self.effect_id, new_name)
+        rule = self.teacher.get_rule()
+        assert new_name == rule.parsers[self.effect_id].pattern_name
+
+        self.assertRaises(
+            NotUniqueParserName, self.teacher.set_pattern_name, self.cause1_id, new_name
+        )
+
+    def test_setting_converter(self):
+        parser = self.teacher.get_rule().parsers[self.cause2_id]
+        new_converter = ConverterType.TO_FLOAT
+        assert not new_converter == parser.groups[3].converter
+
+        self.teacher.set_converter(self.cause2_id, 3, new_converter)
+        parser = self.teacher.get_rule().parsers[self.cause2_id]
+        assert new_converter == parser.groups[3].converter
+
+    def test_setting_primary_key(self):
+        parser = self.teacher.get_rule().parsers[self.cause1_id]
+        new_primary_key_groups = [1, 2]
+        assert not new_primary_key_groups == parser.primary_key_groups
+        self.teacher.set_primary_key(self.cause1_id, new_primary_key_groups)
+        parser = self.teacher.get_rule().parsers[self.cause1_id]
+        assert new_primary_key_groups == parser.primary_key_groups
+
+    def test_setting_log_type(self):
+        new_log_type = '_very_low_likely_log_type'
+        self.teacher.set_log_type(self.effect_id, new_log_type)
+        parser = self.teacher.get_rule().parsers[self.effect_id]
+        assert new_log_type == parser.log_type_name
+
+    def test_update_pattern(self):
+        new_effect_pattern = r'^([0-9]{4}-[0-9]{1,2}-[0-9]{1,2} [0-9]{1,2}:[0-9]{1,2}:[0-9]{1,2}) Error occurred in (.*) test$'
+        self.teacher.update_pattern(self.effect_id, new_effect_pattern)
+        updated_pattern = self.teacher.get_rule().parsers[self.effect_id].pattern
+        assert new_effect_pattern == updated_pattern
+
+        not_matching_pattern = new_effect_pattern + 'not_matching_part_of_regex'
+        self.assertRaises(
+            NotMatchingRegexError, self.teacher.update_pattern, self.effect_id, not_matching_pattern
```

Comment:
I'm not sure how to handle exceptions like that. I thought that teacher.verify method should inform about that.